common/forks/forks.go

```diff
@@ -1,37 +1,54 @@
 package forks
 
 import (
+	"math"
 	"math/big"
 	"sort"
 
 	"github.com/scroll-tech/go-ethereum/params"
 )
 
 // CollectSortedForkHeights returns a sorted set of block numbers that one or more forks are activated on
-func CollectSortedForkHeights(config *params.ChainConfig) ([]uint64, map[uint64]bool) {
+func CollectSortedForkHeights(config *params.ChainConfig) ([]uint64, map[uint64]bool, map[string]uint64) {
 	forkHeightsMap := make(map[uint64]bool)
-	for _, fork := range []*big.Int{
-		config.HomesteadBlock,
-		config.DAOForkBlock,
-		config.EIP150Block,
-		config.EIP155Block,
-		config.EIP158Block,
-		config.ByzantiumBlock,
-		config.ConstantinopleBlock,
-		config.PetersburgBlock,
-		config.IstanbulBlock,
-		config.MuirGlacierBlock,
-		config.BerlinBlock,
-		config.LondonBlock,
-		config.ArrowGlacierBlock,
-		config.ArchimedesBlock,
-		config.ShanghaiBlock,
+	forkNameHeightMap := make(map[string]uint64)
+	type nameFork struct {
+		name  string
+		block *big.Int
+	}
+	for _, fork := range []nameFork{
+		{name: "homestead", block: config.HomesteadBlock},
+		{name: "daoFork", block: config.DAOForkBlock},
+		{name: "eip150", block: config.EIP150Block},
+		{name: "eip155", block: config.EIP155Block},
+		{name: "eip158", block: config.EIP158Block},
+		{name: "byzantium", block: config.ByzantiumBlock},
+		{name: "constantinople", block: config.ConstantinopleBlock},
+		{name: "petersburg", block: config.PetersburgBlock},
+		{name: "istanbul", block: config.IstanbulBlock},
+		{name: "muirGlacier", block: config.MuirGlacierBlock},
+		{name: "berlin", block: config.BerlinBlock},
+		{name: "london", block: config.LondonBlock},
+		{name: "arrowGlacier", block: config.ArrowGlacierBlock},
+		{name: "archimedes", block: config.ArchimedesBlock},
+		{name: "shanghai", block: config.ShanghaiBlock},
+		{name: "banach", block: config.BanachBlock},
 	} {
-		if fork == nil {
+		if fork.block == nil {
+			continue
+		}
+
+		height := fork.block.Uint64()
+		if height == 0 {
+			continue
+		}
```

Comment:
If you do this, `getHardForkNumberByName` will return an error.
```suggestion
```

Reply:
this is the original logic 

https://github.com/scroll-tech/scroll/blob/develop/common/forks/forks.go#L32-L34 

I think @omerfirmak don't think the hard from 0, or other consideration？

Reply:
This would mostly happen during testing, where we might set `banachBlock = 0`.

This is not relevant for the "respect fork boundary" task.

But if the prover provides `forkName = "banach"`, we should be able to assign tasks to it, even if `banachBlock = 0`.

Reply:
Changing this now might require revisiting the previous changes we made as well.
 
My thinking was; if a fork is enabled on genesis it is not really a fork but the initial configuration of the chain. So, every actor taking part in the network is expected to support at least the initial configuration. I would say `forkName` should only be used for forks with non-zero heights.

Reply:
Yes, but the problem is that this might differ from network to network. Not just on a test env, if we launch a new instance of Scroll in the future, banach will probably be set at `0`.

How about we store all non-zero forks in `forkNameHeightMap` (current impl), plus we also store the last genesis fork (e.g. `shanghai` on Scroll mainnet) in this mapping?

Reply:
> How about we store all non-zero forks in forkNameHeightMap (current impl), plus we also store the last genesis fork (e.g. shanghai on Scroll mainnet) in this mapping?

Sounds good. We probably need to remove `if _, ok := forkHeightsMap[height]; ok {` check as well

> Yes, but the problem is that this might differ from network to network. Not just on a test env, if we launch a new instance of Scroll in the future, banach will probably be set at 0.

I see, but unless we wan't to share provers between Scroll instances I don't think there will be an issue.

Reply:
as discuss, add the genesis hard fork. [2fafb34](https://github.com/scroll-tech/scroll/pull/1182/commits/2fafb34a7301c382cf9870109a755056f0c08402)